### DISCUSSION TITLE
Using pyvista with pyinstaller

### DISCUSSION THREAD

**Discussion by meengel** [pyvista/pyvista#4919] [Q&A]
## Creating EXE with pyinstaller
Dear Pyvista-Team,

currently I am writing on some python utilities for analysing Geodata. For visualization purposes I want to use pyvista as it seems to be the best/most appropriate solution for me since I am not an expert in vtk or OpenGL.

One method enables the user to view a 3D-model of the chosen raster. Here is an example of the Geisslerspitzen (data kindly provided as part of the open data initiative by the autonomous province of Bozen):
![image](https://user-images.githubusercontent.com/63775634/82311253-bf945d80-99c5-11ea-82f5-34638fd0d7db.png)

As this method is used by a raster-visualization-program I would like to distribute to some others for testing purposes, I am using [pyinstaller](https://pyinstaller.readthedocs.io/en/stable/operating-mode.html) for creating an exe-file. Although the creation finishes successfully it does not run as there seems to be an issue with the import of the vtk-libraries. I tried hidden imports and stuff - it gives different errors then - but it still does not work.
Do you know anything about this issue? Is there some hiddenimport or hook I am missing?

The problem still exists for a very small Hello-World program importing pyvista - even without using it at all:
```python
import pyvista as pv
if __name__ == '__main__':
    print('Hello World!')
```

I am using anaconda3 (version from october 2019), python3.7, pyinstaller3.6 and pyvista 0.24.1. I installed pyinstaller and pyvista both via the conda-forge channel.

Thank you very much in advance!

## Export as obj
Some additional question: I tried to export the scene as an obj-file using `plotter.export_obj(file)` but it does not include the texture in the corresponding MTL-file. Is there a method available for this? I could not find a solution within the documentation or the Internet.

**Comment by akaszynski** [comment]
Hello,

I've had problems getting `pyvista` to freeze with anaconda, but have gotten it to work with vanillia python.  Have you tried getting it to work from vanilla python?  Installing the necessary packages is possible using `pip` instead of `conda install`

**Comment by btguilherme** [comment]
Hello,

I am also having trouble using PyInstaller to generate the executable for my program that uses pyvista.
@akaszynski , could you explain better how you are doing to generate the executable? I use the vanilla version of python and even then the pyvista module is not added.

Thanks in advance.

**Comment by akaszynski** [comment]
@btguilherme, what's your env?  I've always had issues with conda and `pyinstaller`.  Have you tried vanilla python instead of a conda env?

**Comment by btguilherme** [comment]
@akaszynski, I am using pure `python 3.7.4 (64 bits)`, without conda. `PyInstaller 3.5` and `pyvista 0.25.3`.
I use some `--hidden-import` to run the pyinstaller (`--hidden-import=sklearn`, `--hidden-import=scipy`, `--hidden-import=vtk` and `--hidden-import=pyvista`), without success.

**Comment by CristobalBL** [comment]
I use these hidden imports into .spec file, with python 3.7 and PyInstaller 3.6 into a venv (without conda):

hiddenimports=['vtkmodules', 'vtkmodules.all', 'vtkmodules.util.numpy_support', 'vtkmodules.numpy_interface', 'vtkmodules.numpy_interface.dataset_adapter']

And works!

**Comment by albezem** [comment]
I tried it. But I am getting `ModuleNotFoundError: No module named 'vtk.qt'`` upon running. 

> 
> 
> I use these hidden imports into .spec file, with python 3.7 and PyInstaller 3.6 into a venv (without conda):
> 
> hiddenimports=['vtkmodules', 'vtkmodules.all', 'vtkmodules.util.numpy_support', 'vtkmodules.numpy_interface', 'vtkmodules.numpy_interface.dataset_adapter']
> 
> And works!

It seems a hook file should be submitted to the pyInstaller-hooks-contrib (https://github.com/pyinstaller/pyinstaller-hooks-contrib)


It worked finally as I used following hiddenimports
hiddenimports=['vtkmodules', 'vtkmodules.all', 'vtkmodules.util.numpy_support', 'vtkmodules.numpy_interface', 'vtkmodules.numpy_interface.dataset_adapter','vtkmodules.qt', 'vttmodules.util','vttmodules.vtkCommonCore','vttmodules.vtkCommonKitPython','vtkmodules.qt.QVTKRenderWindowInteractor']

**Comment by meengel** [comment]
Thank you very much!
> hiddenimports=['vtkmodules', 'vtkmodules.all', 'vtkmodules.util.numpy_support', 'vtkmodules.numpy_interface', 'vtkmodules.numpy_interface.dataset_adapter','vtkmodules.qt', 'vttmodules.util','vttmodules.vtkCommonCore','vttmodules.vtkCommonKitPython','vtkmodules.qt.QVTKRenderWindowInteractor']

This solved the problem for me!

**Comment by riteshbarman02** [comment]
![Screenshot 2025-01-03 182205](https://github.com/user-attachments/assets/301db416-45d0-4014-958c-84c893d5007a)
how can i solve this error
